src/rpc/ngHandlers/AccountTx.cpp

```diff
@@ -0,0 +1,223 @@
+    if (minIndex > maxIndex)
+        return Error{
+            RPC::Status{RPC::RippledError::rpcINVALID_PARAMS, "invalidIndex"}};
+
+    if (input.ledgerHash || input.ledgerIndex)
+    {
+        // rippled does not have this check
+        if (input.ledgerIndexMax || input.ledgerIndexMin)
+            return Error{RPC::Status{
+                RPC::RippledError::rpcINVALID_PARAMS,
+                "containsLedgerSpecifierAndRange"}};
+
+        auto const lgrInfoOrStatus = RPC::getLedgerInfoFromHashOrSeq(
+            *sharedPtrBackend_,
+            yield,
+            input.ledgerHash,
+            input.ledgerIndex,
+            range->maxSequence);
+
+        if (auto status = std::get_if<RPC::Status>(&lgrInfoOrStatus))
+            return Error{*status};
+
+        maxIndex = minIndex = std::get<ripple::LedgerInfo>(lgrInfoOrStatus).seq;
+    }
+    std::optional<Backend::TransactionsCursor> cursor;
+    // if marker exists
+    if (input.marker)
+    {
+        cursor = {input.marker->ledger, input.marker->seq};
+    }
+    else
+    {
+        if (input.forward)
+            cursor = {minIndex, 0};
+        else
+            cursor = {maxIndex, INT32_MAX};
+    }
+    auto constexpr limitDefault = 50;
+    auto const limit = input.limit ? *input.limit : limitDefault;
+    auto const accountID = RPC::accountFromStringStrict(input.account);
+    auto const [blobs, retCursor] = sharedPtrBackend_->fetchAccountTransactions(
+        *accountID, limit, input.forward, cursor, yield);
+
+    Output response;
+    if (retCursor)
+        response.marker = {
+            retCursor->ledgerSequence, retCursor->transactionIndex};
+
+    for (auto const& txnPlusMeta : blobs)
+    {
+        // over the range
+        if ((txnPlusMeta.ledgerSequence < minIndex && !input.forward) ||
+            (txnPlusMeta.ledgerSequence > maxIndex && input.forward))
+        {
+            response.marker = std::nullopt;
+            break;
+        }
+        else if (txnPlusMeta.ledgerSequence > maxIndex && !input.forward)
+        {
+            gLog.debug() << "Skipping over transactions from incomplete ledger";
+            continue;
+        }
+
+        boost::json::object obj;
+        if (!input.binary)
+        {
+            auto [txn, meta] = RPC::toExpandedJson(txnPlusMeta);
+            obj[JS(meta)] = meta;
+            obj[JS(tx)] = txn;
+            obj[JS(tx)].as_object()[JS(ledger_index)] =
+                txnPlusMeta.ledgerSequence;
+            obj[JS(tx)].as_object()[JS(date)] = txnPlusMeta.date;
+        }
+        else
+        {
+            obj[JS(meta)] = ripple::strHex(txnPlusMeta.metadata);
+            obj[JS(tx_blob)] = ripple::strHex(txnPlusMeta.transaction);
+            obj[JS(ledger_index)] = txnPlusMeta.ledgerSequence;
+            // only clio has this field
+            obj[JS(date)] = txnPlusMeta.date;
+        }
+        obj[JS(validated)] = true;
+        response.transactions.push_back(obj);
+    }
+
+    if (input.limit)
```

Comment:
The original logic is : when there is no limit in input , we don't have limit in output.

Reply:
since `output.limit` is `std::optional` this logic would remain the same if you just copy the value over (e.g. if `input.limit` is nullopt => `output.limit` will be too). I don't mind if you leave it as is either - it's very minor.